src/Raven.Server/Documents/DocumentDatabase.cs

```diff
@@ -1507,20 +1507,23 @@ public void StateChanged(long index)
                     record = _serverStore.Cluster.ReadDatabase(context, Name);
                 }
 
-                if (_lastTopologyIndex < record.Topology.Stamp.Index)
+                if (ServerStore.HasDatabaseRecordTopologyChanged(record.Topology.Stamp.Index, _lastTopologyIndex, out string url))
                 {
-                    var clusterTopology = ServerStore.GetClusterTopology();
-                    var url = clusterTopology.GetUrlFromTag(ServerStore.NodeTag);
-                    if (url != null)
+                    _lastTopologyIndex = record.Topology.Stamp.Index;
+
+                    Changes.RaiseNotifications(new TopologyChange
                     {
-                        _lastTopologyIndex = record.Topology.Stamp.Index;
+                        Url = url,
+                        Database = Name
+                    });
 
-                        Changes.RaiseNotifications(new TopologyChange
-                        {
-                            Url = url,
-                            Database = Name
-                        });
-                    }
+                    _ = RequestExecutor.UpdateTopologyAsync(new RequestExecutor.UpdateTopologyParameters(new ServerNode()
+                    {
+                        ClusterTag = _serverStore.NodeTag, Database = Name, Url = _serverStore.Server.WebUrl
```

Comment:
I think `WebUrl` is wrong here what if we start the server with `0.0.0.0` ?
Probably better to use `GetNodeHttpServerUrl`

Reply:
Changed + changed in all other places